yb-voyager/cmd/exportSchemaOptimizationReport.go

```diff
@@ -316,10 +361,12 @@ func buildRedundantIndexChange(indexTransformer *sqltransformer.IndexFileTransfo
 	return NewRedundantIndexChange(!bool(skipPerfOptimizations), redundantIndexesFile, getTableToIndexMap(redundantIndexes))
 }
 
-func buildColocationTableRecommendationChange(shardedTables []string, colocatedTables []string) *ColocationRecommendationChange {
+func buildColocationTableRecommendationChange(shardedTables []string, colocatedTables []string, isMigrationAssessmentDoneDirectly bool) *ColocationRecommendationChange {
 	if !assessmentRecommendationsApplied { //If assessment recommendations not applied and skip recommendations is true, then show that its not applied
 		if skipRecommendations {
 			return NewAppliedColocationRecommendationChange("", false, "", nil, nil) // Dummy entry for both table and mview as no need to show two
+		} else if !isMigrationAssessmentDoneDirectly {
```

Comment:
I feel like this logic should not be duplicated here. The transformer layer should decide what to apply and what not to apply. The report generation layer should just generate the report. 

Maybe in some sitiuations like here, that may not be entirely possible, but let's discuss this once on call.